xscen/catalog.py

```diff
@@ -347,6 +347,77 @@ def exists_in_cat(self, **columns):
             logger.info(f"An entry exists for: {columns}")
         return exists
 
+    def to_dataset(
```

Comment:
Could you write some examples because this is not performing the way I am expecting from the docstring?
 
 2 examples of it not doing what I expected:
 
 1. `cat_with_4_experiments.to_dataset(concat_on=['experiment'])` I am expecting this to work and have a coordinate 'experiment'.
 
 What I get:  `ValueError: Expected exactly one dataset. Received 4 datasets.`
 
 2. `cat_with_4_experiments_and_3_members.to_dataset(ensemble_on=[ 'experiment'])`. I am expecting this to fail because we have too many members.
 
 What I get: I works and returns a dataset with coordinate `realization` with the 4 experiments. Unclear which member was used.

Reply:
Hum woups, I think this is because I don't handle the `id` correctly below. Will change and make more tests.

Reply:
Last commit should have fixed the issue.

```python3
import xscen as xs
cat = xs.DataCatalog('catalogues/simulation.json')

scat = cat.search(mip_era='CMIP6', variable='tas', source='FGOALS.*', activity='ScenarioMIP')
scat.to_dataset(concat_on='experiment', ensemble_on='source')
```
works

Reply:
Ok example 1 works as expected now!
But, example 2 still doesn't fail...
With the simulationc catalogue:
`cat.search(source='MIROC6', xrfreq='D', activity='ScenarioMIP').to_dataset(ensemble_on=['experiment'])`

The subcat has 12 datasets (3 members x 4 experiments), but the output only has 4 datasets inside it (the experiments). I would have expected this to fail...

Reply:
Ok, so my ID editing logic was not good. I instead _added_ a `realization` column to concat on and edited the ID by removing all concatenated files. It should now work!